### PR TITLE
Fixed #29942 -- Fixed viewsource links to contrib.auth.forms not appearing.

### PR DESCRIPTION
**Pull request by hramezani**
[Ticket 29942](https://code.djangoproject.com/ticket/29942)

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -102,6 +102,22 @@ def django_release():
...
+    pass
+else:
+    settings.configure()
+    settings.INSTALLED_APPS = [
+        'django.contrib.admin',
+        'django.contrib.auth',
+        'django.contrib.contenttypes',
+        'django.contrib.sessions',
+    ]
+    django.setup()
```
Comment (felixxm) [inline docs/conf.py]: I don't think that we need to call `django.setup()`. I would prefer to load only necessary apps, i.e.
```python
try:
    from django.conf import settings
    from django.apps import apps
except ImportError:
    pass
else:
    settings.configure()
    apps.populate(['django.contrib.contenttypes', 'django.contrib.auth'])

```

Comment (hramezani) [reply docs/conf.py]: @felixxm Thanks for the review.
As I [commented](https://code.djangoproject.com/ticket/29942#comment:3) on the ticket. there is also some missed source page from `session` and `admin` apps.
If we don't populate these apps we will miss the following source files:

- django/contrib/admin/models.html
- django/contrib/admin/apps.html
- django/contrib/sessions/base_session.html

I don't know the exact location of the source link for these there files, I found them by making the diff between files with your patch and my old patch.

If you want to know the exact location of them I can find them.

**Comment by hramezani** [comment]
@felixxm still has an error on `docs` tests.
`django.core.exceptions.ImproperlyConfigured: 'django.db.backends.dummy' isn't an available database backend.
Try using 'django.db.backends.XXX', where XXX is one of:
    'mysql', 'oracle', 'postgresql', 'sqlite3'`

**Comment by felixxm** [comment]
@hramezani Ahh, yes :disappointed: I think it goes into the wrong direction. A full Django configuration shouldn't be require to build docs. We can try to use [viewcode-find-source](https://www.sphinx-doc.rg/en/master/usage/extensions/viewcode.html#event-viewcode-find-source):

> _"In addition, if you don’t want to import the modules by viewcode, you can tell the location of the location of source code to viewcode using the viewcode-find-source event."_

**Comment by felixxm** [comment]
Closing due to inactivity. Please feel-free to send a new PR.

**Pull request closed without merging**